### PR TITLE
Move from JSCombiner to webpack: merged-share-backend

### PR DESCRIPTION
**Pull request by rullzer**
Basically this moved the share backend js code from the JSCombiner to webpack. It still works (at least for me).

A simple `npm run build` does all the magic.

Advantages I see:
1. smaller (111kb vs 203kb)
2. webserver can server the file no need for Nextcloud to get invovled
3. a map file to actually make debugging not a pain
4. With babel we can make use of fancy new js features if needed

CC: @ChristophWurst @skjnldsv @juliushaertl

### CODE REVIEW

File: core/webpack.common.js [hunk 1]
```diff
@@ -1,51 +1,81 @@
...
-			jQuery: "jquery"
-		})
-	],
-	resolve: {
-		alias: {
-			handlebars: 'handlebars/runtime'
+	{
+		entry: {
+			share_backend: [
+				path.resolve(__dirname, 'js/shareconfigmodel.js'),
```
Comment (juliushaertl) [inline core/webpack.common.js]: Any reason to not use a simple entry file and import all those scripts? Seems a bit cleaner to me than having all those in the webpack config.

Comment (rullzer) [reply core/webpack.common.js]: The reason mainly that my webpack fu is under developed.

Comment (rullzer) [reply core/webpack.common.js]: done

File: core/js/merged-share-backend.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+require('./shareconfigmodel.js')
```
Comment (ChristophWurst) [inline core/js/merged-share-backend.js]: on a general note: we should get rid of the legacy module patterns. when we have webpack like here, we can make those files real modules and just assign them to OC.Share here (which right now is done at https://github.com/nextcloud/server/blob/b58f3e75839669d5b0bf3c3b2f8ee950d8accab1/core/js/shareconfigmodel.js#L71).

Of course that can be refactored after this PR.

**Review by ChristophWurst — APPROVED** [review]
Tested, works :+1: I like it :+1: 

The only think I would like to see changed (but I'm fine with a follow-up PR) is to move scripts that are loaded webpack to `core/src`. That will make it easier for us to distinguish between new and old modules. I can do that together with the other refactoring I suggested.

**Comment by rullzer** [comment]
Ready to get in :)

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Btw, any reasons to use require and not import?
@ChristophWurst @juliushaertl is there any differences in that? :thinking:

**Comment by ChristophWurst** [comment]
We should use import. I think it's not a problem if you don't mix the two in the same bundle.

**Pull request merged**